modules/mauth-signer-http4s/src/main/scala/com/mdsol/mauth/http4s/client/Implicits.scala

```diff
@@ -0,0 +1,46 @@
+package com.mdsol.mauth.http4s.client
+
+import com.mdsol.mauth.models.SignedRequest
+import org.http4s.headers.`Content-Type`
+import org.http4s.{headers, Header, Headers, Method, Request, Uri}
+import org.typelevel.ci.CIString
+
+import scala.annotation.nowarn
+import scala.collection.immutable
+
+object Implicits {
+
+  implicit class NewSignedRequestOps(val signedRequest: SignedRequest) extends AnyVal {
+
+    /** Create an akka-http request from a [[models.SignedRequest]]
+      */
+    def toHttp4sRequest[F[_]]: Request[F] = {
+      val contentType: Option[`Content-Type`] = extractContentTypeFromHeaders(signedRequest.req.headers)
+      val headersWithoutContentType: Map[String, String] = removeContentTypeFromHeaders(signedRequest.req.headers)
+
+      val allHeaders: immutable.Seq[Header.Raw] = (headersWithoutContentType ++ signedRequest.mauthHeaders).toList
+        .map { case (name, value) =>
+          Header.Raw(CIString(name), value)
+        }
+
+      Request[F](
+        method = Method.fromString(signedRequest.req.httpMethod).getOrElse(Method.GET),
+        uri = Uri(path = Uri.Path.unsafeFromString(signedRequest.req.uri.toString)),
+        body = fs2.Stream.emits(signedRequest.req.body),
+        headers = Headers(allHeaders)
+      ).withContentTypeOption(contentType)
+    }
+
+    private def extractContentTypeFromHeaders(requestHeaders: Map[String, String]): Option[`Content-Type`] =
+      requestHeaders
+        .get(headers.`Content-Type`.toString)
```

Comment:
I'd claim you shouldn't be using `headers.`Content-Type`` from `http4s` is the header you're trying to intercept is foreign to it

Reply:
here we are just extracting the content type header, so I don't think it's doing anything foreign

Reply:
The header types though arent the http4s headers, they're just strings. I think its ok though as its just a constant you'd have to define again.

Reply:
Sorry, I think my comment was really unclear. What I mean is that
1. `requestHeaders` is a Map of String representations of non-http4s headers
2. we're asking to get us the value of the `Content-Type` header based on the string representation of http4s
3. because of that you might end up with a mismatch.
Now, this is a minor thing, so please don't get blocked on this. Just wanted to clarify what I meant :-)